### PR TITLE
* added path option to rebase the destination path of the written/cop…

### PR DESCRIPTION
**Pull request by canercandan**
…ied files + unit tests

PR to fix the issue discussed on yeoman/generator#872, all tests green.

### CODE REVIEW

File: generators/app/index.js [hunk 1]
```diff
@@ -11,18 +11,30 @@ function sortByKeys(a, b) {
...
+      desc: 'Use an other destination path.'
+    });
+  },
+
   writing: {
     app: function () {
       var optional =  this.options.config || {};
-      var existing = this.fs.exists(this.destinationPath('.travis.yml'))
-            ? yaml.parse(this.fs.read(this.destinationPath('.travis.yml')))
+      var existing = this.fs.exists(this.destinationPath(this.options.path + '.travis.yml'))
```
Comment (SBoudrias) [inline generators/app/index.js]: The node way of concatenating path is `path.join(a, b, c?)`.

Using `+`, you're relying on the user to provide a `/` which will not always be the case.

Comment (SBoudrias) [inline generators/app/index.js]: Oh also, pretty sure Yeoman utils do it for you, so just `this.destinationPath(this.options.path, '.travis.yml')`

**Comment by canercandan** [comment]
My apologize I applied the same instructions I saw on https://github.com/yeoman/generator-node#publishing-your-code

**Comment by iamstarkov** [comment]
its okay, that instructutions are mainly for maintainers. another problem with version bump in pull-request that it will not include git tags

**Comment by canercandan** [comment]
I will create a new PR without the version upgrade. Is there anything else you want me to change ?

**Comment by iamstarkov** [comment]
no need to create another pr, just revert last commit

**Comment by iamstarkov** [comment]
it will be just:

```
git revert 1f4d220
```

**Comment by canercandan** [comment]
done.

**Comment by iamstarkov** [comment]
about your last response in the yeoman/generator#872, does it make sense it context of travis configs, because `.travis.yml` have to be in the root of the project

**Comment by iamstarkov** [comment]
obviously, I get the point and also can see other applications as well, and will merge this pr later today

**Comment by iamstarkov** [comment]
asking about "fullstack" case in context of generator-travis, because want to understand will it help you or not

**Comment by canercandan** [comment]
@iamstarkov you are right on this, there is no support for multiple .travis.yml files this issue can confirm https://github.com/travis-ci/travis-ci/issues/3540

**Comment by iamstarkov** [comment]
okay, good. but I still think it can be useful in the case if someone will create generator-batch-update which will update all projects you have in for example `~/projects`

**Comment by iamstarkov** [comment]
what do you think?

**Comment by canercandan** [comment]
fully agree with you :+1:

**Comment by canercandan** [comment]
@iamstarkov  for consistency matter, are you ok I replace the `path` option by `generate-into` as discussed in https://github.com/yeoman/generator-node/pull/186 ?

**Comment by iamstarkov** [comment]
yep, its fine

**Comment by canercandan** [comment]
@iamstarkov done, all tests green

**Comment by canercandan** [comment]
commits have been squashed too

**Comment by SBoudrias** [comment]
@canercandan only two files are modified now?

**Comment by canercandan** [comment]
yes since I removed the commits that upgrade the version and the one who revert it ;-)

**Comment by SBoudrias** [comment]
@canercandan ah my bad, I mistakenly took that one for the one on generator-node

**Comment by canercandan** [comment]
@SBoudrias ok

**Comment by iamstarkov** [comment]
@canercandan looked into to the code, looks good.

**Comment by iamstarkov** [comment]
@canercandan @SBoudrias one more small question before merge. why its not enough to add it to only composed generators, like generator-node? I guess yeoman doesnt overwrite `this.destinationPath` for nested generators, so even generator-node get support for `generateInto` option it still has no effect here, am i right?

**Comment by SBoudrias** [comment]
@iamstarkov yes, there's no effect on sub-generators.

We might add it as a feature eventually: https://github.com/yeoman/generator/issues/872

**Comment by canercandan** [comment]
@iamstarkov since I am refactoring my package.json, I wanted to know if you wanted to merge this PR anytime soon. Thanks.

**Comment by iamstarkov** [comment]
of course

**Comment by iamstarkov** [comment]
we need to update README to be able to publish new versions with your pr

**Comment by iamstarkov** [comment]
I will try do so today

**Comment by canercandan** [comment]
perfect, thanks ;-)

**Pull request merged**